core/src/trezor/crypto/slip39.py

```diff
@@ -0,0 +1,606 @@
+        0xE0E0009,
+        0x1C0C2412,
+        0x38086C24,
+        0x3090FC48,
+        0x21B1F890,
+        0x3F3F120,
+    )
+    chk = 1
+    for v in values:
+        b = chk >> 20
+        chk = (chk & 0xFFFFF) << 10 ^ v
+        for i in range(10):
+            chk ^= GEN[i] if ((b >> i) & 1) else 0
+    return chk
+
+
+def rs1024_create_checksum(data):
+    values = tuple(CUSTOMIZATION_STRING) + data + CHECKSUM_LENGTH_WORDS * (0,)
+    polymod = _rs1024_polymod(values) ^ 1
+    return tuple(
+        (polymod >> 10 * i) & 1023 for i in reversed(range(CHECKSUM_LENGTH_WORDS))
+    )
+
+
+def rs1024_verify_checksum(data):
+    return _rs1024_polymod(tuple(CUSTOMIZATION_STRING) + data) == 1
+
+
+def xor(a, b):
+    return bytes(x ^ y for x, y in zip(a, b))
+
+
+def _int_from_indices(indices):
+    """Converts a list of base 1024 indices in big endian order to an integer value."""
+    value = 0
+    for index in indices:
+        value = (value << RADIX_BITS) + index
+    return value
+
+
+def _int_to_indices(value, length, bits):
+    """Converts an integer value to indices in big endian order."""
+    mask = (1 << bits) - 1
+    return ((value >> (i * bits)) & mask for i in reversed(range(length)))
+
+
+def mnemonic_from_indices(indices):
+    return " ".join(wordlist[i] for i in indices)
+
+
+def mnemonic_to_indices(mnemonic):
+    return (word_index(word.lower()) for word in mnemonic.split())
+
+
+def _round_function(i, passphrase, e, salt, r):
+    """The round function used internally by the Feistel cipher."""
+    return pbkdf2(
+        pbkdf2.HMAC_SHA256,
+        bytes([i]) + passphrase,
+        salt + r,
+        (BASE_ITERATION_COUNT << e) // ROUND_COUNT,
+    ).key()[: len(r)]
+
+
+def _get_salt(identifier):
+    return CUSTOMIZATION_STRING + identifier.to_bytes(
+        bits_to_bytes(ID_LENGTH_BITS), "big"
+    )
+
+
+def _encrypt(master_secret, passphrase, iteration_exponent, identifier):
+    l = master_secret[: len(master_secret) // 2]
+    r = master_secret[len(master_secret) // 2 :]
+    salt = _get_salt(identifier)
+    for i in range(ROUND_COUNT):
+        (l, r) = (
+            r,
+            xor(l, _round_function(i, passphrase, iteration_exponent, salt, r)),
+        )
+    return r + l
+
+
+def decrypt(identifier, iteration_exponent, encrypted_master_secret, passphrase):
+    l = encrypted_master_secret[: len(encrypted_master_secret) // 2]
+    r = encrypted_master_secret[len(encrypted_master_secret) // 2 :]
+    salt = _get_salt(identifier)
+    for i in reversed(range(ROUND_COUNT)):
+        (l, r) = (
+            r,
+            xor(l, _round_function(i, passphrase, iteration_exponent, salt, r)),
+        )
+    return r + l
+
+
+def _create_digest(random_data, shared_secret):
+    return hmac.new(random_data, shared_secret, hashlib.sha256).digest()[
+        :DIGEST_LENGTH_BYTES
+    ]
+
+
+def _split_secret(threshold, share_count, shared_secret):
+    assert 0 < threshold <= share_count <= MAX_SHARE_COUNT
```

Comment:
This is the only assertion in this file. I suggest throwing an exception instead.